Changing the number of jobs in a cv report creates a recursion error in `metrics.summarize()`

### Describe the bug

Title
The weird part is that it does not prevent the docs from building in #3145 even though the bugged code was extracted from there.

### Steps/Code to Reproduce

Using the test fixtures:
```python
estimator, X, y = forest_binary_classification_data
report = CrossValidationReport(estimator, X=X, y=y, splitter=5, n_jobs=5)
report.metrics.summarize()
```
#3175 shows this crash on the CI.

### Expected Behavior

No crash

### Actual Behavior

```bash
  File "/home/runner/work/skore/skore/skore/venv/lib/python3.11/site-packages/skore/_sklearn/_base.py", line 190, in __getattr__
    if name in self.available():
               ^^^^^^^^^^^^^^^^
  File "/home/runner/work/skore/skore/skore/venv/lib/python3.11/site-packages/skore/_sklearn/_estimator/metrics_accessor.py", line 257, in available
    return list(self._parent._metric_registry)
                ^^^^^^^^^^^^
  File "/home/runner/work/skore/skore/skore/venv/lib/python3.11/site-packages/skore/_sklearn/_base.py", line 190, in __getattr__
    if name in self.available():
               ^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded
"""

The above exception was the direct cause of the following exception:

forest_binary_classification_data = (RandomForestClassifier(), array([[-1.05383855, -1.02754411, -0.32929388,  0.82600732],
       [ 1.56931739,  1.306542...,
       0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0,
       1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1]))

    def test_non_default_n_jobs(forest_binary_classification_data):
        estimator, X, y = forest_binary_classification_data
        report = CrossValidationReport(estimator, X=X, y=y, splitter=5, n_jobs=5)
>       report.metrics.summarize()

tests/unit/reports/cross_validation/metrics/test_summarize.py:347: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
venv/lib/python3.11/site-packages/skore/_sklearn/_cross_validation/metrics_accessor.py:99: in summarize
    return self._summarize_display(data_source=data_source, metric=metric)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
venv/lib/python3.11/site-packages/skore/_sklearn/_cross_validation/metrics_accessor.py:126: in _summarize_display
    summaries = list(
venv/lib/python3.11/site-packages/skore/_utils/_progress_bar.py:83: in track
    for value in sequence:
venv/lib/python3.11/site-packages/joblib/parallel.py:1682: in _get_outputs
    yield from self._retrieve()
venv/lib/python3.11/site-packages/joblib/parallel.py:1784: in _retrieve
    self._raise_error_fast()
venv/lib/python3.11/site-packages/joblib/parallel.py:1859: in _raise_error_fast
    error_job.get_result(self.timeout)
venv/lib/python3.11/site-packages/joblib/parallel.py:758: in get_result
    return self._return_or_raise()
           ^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <joblib.parallel.BatchCompletionCallBack object at 0x7f76f580c890>

    def _return_or_raise(self):
        try:
            if self.status == TASK_ERROR:
>               raise self._result
E               joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed to un-serialize. Please ensure that the arguments of the function are all picklable.

venv/lib/python3.11/site-packages/joblib/parallel.py:773: BrokenProcessPool
```

### Environment

```shell
All the ci envs
```

--- a/skore/src/skore/_sklearn/_base.py
+++ b/skore/src/skore/_sklearn/_base.py
@@ -26,6 +26,9 @@ if TYPE_CHECKING:
     import pandas as pd
 
     from skore._sklearn._checks.accessor import _ChecksAccessor
+    from skore._sklearn._cross_validation.report import CrossValidationReport
+    from skore._sklearn._estimator.report import EstimatorReport
+    from skore._sklearn._plot import MetricsSummaryDisplay
     from skore._utils.repr.data import AccessorHelpData
 
 
@@ -177,6 +180,20 @@ class _BaseAccessor(AccessorHelpMixin, Generic[ParentT]):
 
     def _repr_mimebundle_(self, **kwargs):
         return {"text/plain": repr(self), "text/html": self._repr_html_()}
+
+
+def _summarize_report_metrics(
+    report: EstimatorReport | CrossValidationReport,
+    *,
+    data_source: DataSource | Literal["both"],
+    metric: str | list[str] | None = None,
+) -> MetricsSummaryDisplay:
+    """Compute a metrics summary for ``report``.
+
+    Pure Python function to avoid pickling a metrics accessor as a bound method's
+    ``__self__``.
+    """
+    return report.metrics._summarize_display(data_source=data_source, metric=metric)
 
 
 class BaseMetricsAccessor(_BaseAccessor, Generic[ParentT]):

--- a/skore/src/skore/_sklearn/_comparison/metrics_accessor.py
+++ b/skore/src/skore/_sklearn/_comparison/metrics_accessor.py
@@ -4,13 +4,13 @@ import numbers
 import warnings
 from typing import Any, Literal
 
-import joblib
 import pandas as pd
+from joblib import Parallel
 from numpy.typing import ArrayLike
 from sklearn.utils.metaestimators import available_if
 
 from skore._externals._pandas_accessors import DirNamesMixin
-from skore._sklearn._base import BaseMetricsAccessor
+from skore._sklearn._base import BaseMetricsAccessor, _summarize_report_metrics
 from skore._sklearn._comparison.report import ComparisonReport
 from skore._sklearn._plot.metrics import (
     ConfusionMatrixDisplay,
@@ -26,6 +26,7 @@ from skore._utils._accessor import (
     _check_supported_ml_task,
 )
 from skore._utils._fixes import _validate_joblib_parallel_params
+from skore._utils._parallel import delayed
 from skore._utils._progress_bar import track
 
 DataSource = Literal["test", "train", "both"]
@@ -84,7 +85,7 @@ class _MetricsAccessor(BaseMetricsAccessor[ComparisonReport], DirNamesMixin):
         precision              0.98...              0.98...
         recall                 0.92...              0.92...
         """
-        parallel = joblib.Parallel(
+        parallel = Parallel(
             **_validate_joblib_parallel_params(
                 n_jobs=self._parent.n_jobs, return_as="generator"
             )
@@ -93,7 +94,8 @@ class _MetricsAccessor(BaseMetricsAccessor[ComparisonReport], DirNamesMixin):
         summaries = list(
             track(
                 parallel(
-                    joblib.delayed(report.metrics._summarize_display)(
+                    delayed(_summarize_report_metrics)(
+                        report,
                         data_source=data_source,
                         metric=metric,
                     )

--- a/skore/src/skore/_sklearn/_cross_validation/metrics_accessor.py
+++ b/skore/src/skore/_sklearn/_cross_validation/metrics_accessor.py
@@ -9,7 +9,7 @@ from numpy.typing import ArrayLike
 from sklearn.utils.metaestimators import available_if
 
 from skore._externals._pandas_accessors import DirNamesMixin
-from skore._sklearn._base import BaseMetricsAccessor
+from skore._sklearn._base import BaseMetricsAccessor, _summarize_report_metrics
 from skore._sklearn._cross_validation.report import CrossValidationReport
 from skore._sklearn._plot import (
     ConfusionMatrixDisplay,
@@ -126,7 +126,8 @@ class _MetricsAccessor(BaseMetricsAccessor[CrossValidationReport], DirNamesMixin
         summaries = list(
             track(
                 parallel(
-                    delayed(report.metrics._summarize_display)(
+                    delayed(_summarize_report_metrics)(
+                        report,
                         data_source=data_source,
                         metric=metric,
                     )

--- a/skore/tests/unit/reports/comparison/test_metrics.py
+++ b/skore/tests/unit/reports/comparison/test_metrics.py
@@ -126,3 +126,30 @@ def test_available_with_unknown_report_name_raises(report):
     comparison_report = ComparisonReport(reports)
     with pytest.raises(ValueError, match="Unknown report name"):
         comparison_report.metrics.available(report_name="unknown")
+
+
+def test_non_default_n_jobs():
+    """summarize() must work when ComparisonReport uses process-based parallelism.
+
+    Joblib must not pickle a bound metrics-accessor method; that used to recurse
+    while unpickling ``__getattr__`` / ``available()`` in worker processes.
+    """
+    X, y = make_classification(random_state=0)
+    reports = {
+        "LinearSVC": EstimatorReport(
+            LinearSVC(), X_train=X, X_test=X, y_train=y, y_test=y, pos_label=1
+        ),
+        "LogisticRegression": EstimatorReport(
+            LogisticRegression(),
+            X_train=X,
+            X_test=X,
+            y_train=y,
+            y_test=y,
+            pos_label=1,
+        ),
+    }
+    comparison_report = ComparisonReport(reports, n_jobs=2)
+    display = comparison_report.metrics.summarize()
+
+    assert isinstance(display, MetricsSummaryDisplay)
+    assert set(display.summary["estimator"]) == {"LinearSVC", "LogisticRegression"}

--- a/skore/tests/unit/reports/cross_validation/metrics/test_summarize.py
+++ b/skore/tests/unit/reports/cross_validation/metrics/test_summarize.py
@@ -339,3 +339,17 @@ def test_pos_label_overwrite(metric, logistic_binary_classification_data):
 
     assert len(display.summary) == 2  # One line per split
     assert display.summary["label"].isna().all()
+
+
+def test_non_default_n_jobs(forest_binary_classification_data):
+    """summarize() must work when the report uses process-based parallelism.
+
+    Joblib must not pickle a bound metrics-accessor method; that used to recurse
+    while unpickling ``__getattr__`` / ``available()`` in worker processes.
+    """
+    estimator, X, y = forest_binary_classification_data
+    report = CrossValidationReport(estimator, X=X, y=y, splitter=5, n_jobs=2)
+    display = report.metrics.summarize()
+
+    assert isinstance(display, MetricsSummaryDisplay)
+    assert set(display.summary["split"]) == {0, 1, 2, 3, 4}

